core/domain/feedback_domain.py

```diff
@@ -362,6 +365,58 @@ def to_dict(self) -> FeedbackMessageReferenceDict:
             'message_id': self.message_id
         }
 
+    @classmethod
+    def from_dict(
+        cls,
+        reference_dict: FeedbackMessageReferenceDict
+        ) -> FeedbackMessageReference:
+        """Return a FeedbackMessageReference domain object from a dict.
+
+        Args:
+            reference_dict: dict. The dict representation of a
+                FeedbackMessageReference object.
+
+        Returns:
+            FeedbackMessageReference. The corresponding FeedbackMessageReference
+            domain object.
+        """
+        feedback_properties = [
+            'entity_type', 'entity_id',
+            'thread_id', 'message_id']
+        for feedback_property in feedback_properties:
+            if feedback_property not in reference_dict:
+                raise utils.ValidationError(
+                    '%s not in reference dict.' % (
+                        feedback_property))
```

Comment:
Why do we need to do this? The code below will fail anyway if these do not exist, so we can just use try except.

Reply:
Noted a similar check in `stats_domain.py`.  

In the case of try/except, should the `from_dict` itself throw exception, or any function calling `from_dict`, or both?

Reply:
I think `from_dict` itself.

Reply:
I addressed this in the latest commit.